### PR TITLE
Push sample app to public ECR for use in playground too.

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -21,16 +21,21 @@ jobs:
...
+      - name: Login to ECR
+        uses: docker/login-action@v1
+        with:
+          registry: public.ecr.ews
+      - name: Build docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: sampleapp
+          tags: |
+            public.ecr.aws/u0d6r4y4/aws-otel-go-test-gorilla:${{ github.sha }}
```
Comment (wyTrivail) [inline .github/workflows/main.yml:36]: it's nice to have an alias for the registry.

File: sampleapp/Dockerfile [hunk 2]
```diff
@@ -1,9 +1,16 @@
...
 WORKDIR /usr/src/app
 
 COPY . .
 
-RUN cd sampleapp && go build
+RUN CGO_ENABLED=0 go build -a -ldflags '-extldflags "-static"'
+
+FROM scratch
+COPY --from=build-env /usr/src/app/sampleapp /sampleapp
+COPY --from=build-env /etc/passwd /etc/passwd
```
Comment (wyTrivail) [inline sampleapp/Dockerfile:13]: what's the reason to copy /etc/passwd?

**Review by wyTrivail — APPROVED** [review]

**Review by bhautikpip — APPROVED** [review]
LGTM :)

File: .github/workflows/main.yml (hunk 1 above)
Comment (anuraaga) [reply .github/workflows/main.yml:36]: Yup - would probably move these to the abs-observability account but for now wanted to start simple / non-controversial :P

File: sampleapp/Dockerfile (hunk 2 above)
Comment (anuraaga) [reply sampleapp/Dockerfile]: Oops copy-pasted, removed

**Pull request merged**